unchained-commons/src/main/java/unchained/commons/contract/Factory.java

```diff
@@ -0,0 +1,22 @@
+package unchained.commons.contract;
+
+import unchained.commons.utils.ReflectionUtils;
+
+import java.util.function.Function;
+
+@FunctionalInterface
+public interface Factory<T> extends Function<Object[], T> {
+
+    T create(Object... arguments);
+
+    @Override
+    default T apply(Object[] arguments) {
```

Comment:
Overriding `Function.apply(...)` and rerouting to a similar function is a functional anti-pattern.